pkg/syslog/rfc6587/writer.go

```diff
@@ -0,0 +1,18 @@
+package rfc6587
+
+import (
+	"fmt"
+	"io"
+)
+
+type Writer struct {
+	w io.Writer
+}
+
+func NewWriter(w io.Writer) *Writer {
+	return &Writer{w}
+}
+
+func (w *Writer) Write(p []byte) (int, error) {
+	return fmt.Fprintf(w.w, "%d %s", len(p), p)
```

Comment:
these `Writer` structs feel like a heavy way to get what you need. 18 lines of code and a new file here just for a single `fmt` call?